enhancements/etcd/etcd-size-tuning.md

```diff
@@ -0,0 +1,230 @@
+* For a larger database, defragmentation of etcd will take longer which will increase the amount of time that etcd is unavaiable for writes.
+  Etcd will still be available for reads during defragmentation, so there should be little impact on the apiserver's availability.
+* Downgrades may be impacted if:
+  1. The downgrade is to a version without this feature.
+  2. The quota has been set higher than the default.
+  3. The etcd backends are using more space than the default.
+  A mitigation would be to keep the quota at the higher value (with no way of decreasing it), so that the cluster will continue to function and no data will be lost.
+* Currently, etcd can not detect whether its backend quota configuration value could lead to a database that's larger than the available storage.
+  * We could potentially add a prometheus alert to fire when the etcd node is running out of disk space.
+  * We could potentially add some validation to the etcd server to check the quota configuration value against the space available and error if too small.
+    * This enhancement would need to take into consideration multiple instances sharing a disk to reduce the likelihood of each individual instance thinking it has enough space, but not realizing the other instances are also vying for that space.
+
+### Drawbacks
+
+* There will be a required etcd rollout when changing values, this happens automatically by the Cluster Etcd Operator to re-render the etcd podspec with the new backend quota value.
+
+## Open Questions [optional]
+
+For each of these, we would need to compare against the current default of 8GiB. Increasing the quota does not necessarily increase the following, but at the extreme case (e.g. increase to 16GiB and using nearly that much) we expect the following to perform differently.
+* How long does it take to sync a snapshot of a given size to another peer; for reboot/recovery?
+* How long does it take to compact & defrag a database of a given size?
+  * Defragmentation will pause writes to etcd, reads from etcd will be unaffected.
+
+## Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+We will need to test a cluster with a larger quota to ensure everything works as expected.
+We should also load the database with a larger quota and make note of the memory requirements of a defragmentation.
+  - We should do this with some values we could assume a customer would use; 16GiB, 32GiB, 100GiB (in the extreme).
+We need to test that it is not possible to shrink the quota once it has be set to a larger value.
+
+## Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+None
+
+### Dev Preview -> Tech Preview
+
+N/A
+
+### Tech Preview -> GA
+
+- Known issue in API Server
+  - When the API server restarts, it's currently possible for all watchers to be reset at the same time and request a large amount of data from etcd, leading to a crash-loop in the API server.
```

Comment:
This was also brought up in the slack, we want to ensure we have a ready procedure for cases when the cluster enters the crash-loop with apiserver. Even after apiserver fixes both bits:
- apistreaming 
- cbor for CRDs
The team will have to create condition to get to the point where we are still able to create that crash-loop and write down a document how to recover from such a state. 

Also I'd like to see both of the above apiserver dependencies (ideally with links to upstream KEPs) being explicitly linked in this proposal, so it's easier to track the current status of those.